sys/x86/include/sigframe.h

```diff
@@ -67,7 +67,11 @@ struct sigframe {
 		__sighandler_t		*sf_handler;
 	} sf_ahu;
 	ucontext_t	sf_uc;		/* = *sf_ucontext */
+#ifdef _KERNEL
+	struct siginfo_native sf_si;	/* = *sf_siginfo (SA_SIGINFO case) */
+#else
```

Comment:
This seems unfortunate.   I'm also guessing that once we have a stable pure cap kernel this will also effectively go away since we will probably be able to drop support for hybrid kernels?

Reply:
The show _native thing is an outgrowth of doing CheriABI as a compat layer.  If we make it the default then we don't need these any more even in hybrid mode (we may need more __kerncap annotations).